osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBall.cs

```diff
@@ -146,24 +147,49 @@ private set
         }
 
         private bool canCurrentlyTrack => Time.Current >= slider.StartTime && Time.Current < slider.EndTime;
+        private OsuAction? trackingAction;
 
         protected override void Update()
         {
             base.Update();
-
             if (Time.Current < slider.EndTime)
             {
                 // Make sure to use the base version of ReceivePositionalInputAt so that we correctly check the position.
                 Tracking = canCurrentlyTrack
                            && lastScreenSpaceMousePosition.HasValue
                            && ReceivePositionalInputAt(lastScreenSpaceMousePosition.Value)
-                           && (drawableSlider?.OsuActionInputManager?.PressedActions.Any(x => x == OsuAction.LeftButton || x == OsuAction.RightButton) ?? false);
+                           && (drawableSlider?.OsuActionInputManager?.PressedActions.Any(x => (x == OsuAction.LeftButton || x == OsuAction.RightButton) && x == trackingAction) ?? false);
             }
+
+            if (!tracking)
+                trackingAction = null;
         }
 
         public void UpdateProgress(double completionProgress)
         {
             Position = slider.CurvePositionAt(completionProgress);
         }
+
+        public bool OnPressed(OsuAction action)
+        {
+            bool hit = false;
+
+            switch (action)
+            {
+                case OsuAction.LeftButton:
+                case OsuAction.RightButton:
+                    hit = IsHovered;
```

Comment:
pretty sure this conditional is gonna be wrong, based on the conditionals on `IsTracking`.